definitions/features/system_repos.rb

```diff
@@ -0,0 +1,48 @@
+class Features::SystemRepos < ForemanMaintain::Feature
+  metadata do
+    label :system_repos
+    description 'Feature for operations on yum repositories of system'
+  end
+
+  def upstream_repos
+    repositories = {}
+    list.each do |repo, url|
+      upstream_repo_urls.each do |regex|
+        repositories[repo] = url if url =~ regex
+      end
+    end
+    return repositories
+  end
+
+  def list
+    repos = execute("yum repolist enabled -d 6 -e 0 2> /dev/null | grep -E 'Repo-id|Repo-baseurl'")
+    return {} if repos.empty?
+
+    Hash[*repos.delete!(' ').split("\n")]
+  end
+
+  def upstream_repos_ids
+    trim_repoids(upstream_repos.keys)
+  end
+
+  def trim_repoids(repos)
```

Comment:
@upadhyeammit Could you make this `private`?

Reply:
done!